cmake-modules/AddCmockaFunctions.cmake

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
We should borrow from https://gitlab.com/cmocka/cmocka/-/blob/master/cmake/Modules/AddCMockaTest.cmake#L1 and modify as needed.

Reply:
I created one from scratch, with our specific needs (private access for tests to az_private, linker flags).  So we should support and maintain our own version.
We don't support wine.